lib/inferno/dsl/fhir_evaluation/rules/all_references_resolve.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative '../../../utils/evaluator_util'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllReferencesResolve < Rule
+          def check(context)
+            # resource_type_ids is for quick look up when there is a reference type
+            # resource_ids is for quick look up when there is no type (i.e. uuid used)
+            util = Inferno::Utils::EvaluatorUtil
+            _, resource_type_ids, resource_ids, references = util.extract_ids_references(context.data)
```

Comment:
You shouldn't be using an array to store diverse pieces of information and relying on the order to determine the meaning of the elements. Use a Hash.

Reply:
Refactored the rules and the reference extractor overall based on the comments.

Reply:
The reference extractor is still putting diverse values in an array and relying on the ordering to determine the meaning.

Reply:
Changed them to either `Hash` or `Set` if values are unique and no need to be ordered.